### DISCUSSION TITLE
SanitizeRxn and working with SMILES based reactions

### DISCUSSION THREAD

**Discussion by vfscalfani** [rdkit/rdkit#7067] [Q&A] [answered]
I am trying to understand when `SanitizeRxn()` should be used. From the docs [1], it seems like the sanitization operations are for reaction templates (e.g., using SMARTS and then running a reaction). However, if I am working with known reactions; that is, loading a reaction with `rdChemReactions.ReactionFromSmarts()` and the `useSMILES=True` option, is there any need or advantage to running `SanitizeRxn()`? 

Thanks for any clarification.

Vin

[1] https://www.rdkit.org/docs/source/rdkit.Chem.rdChemReactions.html#rdkit.Chem.rdChemReactions.SanitizeRxn

**Comment by bp-kelley** [comment, marked as answer]
We just added a sanitize products to SanitizeRxns

https://github.com/rdkit/rdkit/pull/7095


So it might be somewhat useful.   However, you are correct that SanitizeRxns came from the need to 

(1) sanitize aromaticity perception
(2) fix known errors in some RXN files.

that being said, it shouldn't hurt.

**Reply by vfscalfani** [reply]
Hi @bp-kelley. Thanks a lot for your response, this was helpful and answered my question!
